Source/ProjectNL/GAS/Ability/Active/Default/ComboAttack/GA_ComboAttack.cpp

```diff
@@ -130,19 +107,244 @@ void UGA_ComboAttack::ClearDelegate()
 		ComboAttackNotifyState->OnNotifiedBegin.RemoveAll(this);
 		ComboAttackNotifyState->OnNotifiedEnd.RemoveAll(this);
 	}
-	
+
 	if (ComboAttackEndNotify)
 	{
 		ComboAttackEndNotify->OnNotified.RemoveAll(this);
 	}
 }
 
 void UGA_ComboAttack::EndAbility(const FGameplayAbilitySpecHandle Handle
-															, const FGameplayAbilityActorInfo* ActorInfo
-															, const FGameplayAbilityActivationInfo
-															ActivationInfo, bool bReplicateEndAbility
-															, bool bWasCancelled)
+                                 , const FGameplayAbilityActorInfo* ActorInfo
+                                 , const FGameplayAbilityActivationInfo
+                                 ActivationInfo, bool bReplicateEndAbility
+                                 , bool bWasCancelled)
 {
+	if (APlayerController* PlayerController = Cast<APlayerController>(CurrentCharacter->GetController()))
+	{
+		PlayerController->SetIgnoreMoveInput(false);
+		//PlayerController->SetIgnoreLookInput(true);
+		UE_LOG(LogTemp, Warning, TEXT("캐릭터 입력이 활성화 되었습니다."));
+	}
+		
+	UE_LOG(LogTemp, Display, TEXT("Test Combo EndAbility"));
 	Super::EndAbility(Handle, ActorInfo, ActivationInfo, bReplicateEndAbility
-										, bWasCancelled);
-}
\ No newline at end of file
+	                  , bWasCancelled);
+	
+	// 캐릭터 입력 비활성화
+
+}
+
+void UGA_ComboAttack::InputPressed(const FGameplayAbilitySpecHandle Handle, const FGameplayAbilityActorInfo* ActorInfo,
+                                   const FGameplayAbilityActivationInfo ActivationInfo)
+{
+	Super::InputPressed(Handle, ActorInfo, ActivationInfo);
+
+	UE_LOG(LogTemp, Display, TEXT("Test Combo InputPressed"));
+}
+
+void UGA_ComboAttack::InputReleased(const FGameplayAbilitySpecHandle Handle, const FGameplayAbilityActorInfo* ActorInfo,
+                                    const FGameplayAbilityActivationInfo ActivationInfo)
+{
+	Super::InputReleased(Handle, ActorInfo, ActivationInfo);
+	const FTimespan HoldDuration = FDateTime::Now() - InputPressedTime;
+	UAbilitySystemComponent* AbilitySystem = GetAbilitySystemComponentFromActorInfo();
+	if(!AbilitySystem)
+	{
+		return;
+	}
```

Comment:
```suggestion
UAbilitySystemComponent* AbilitySystem = GetAbilitySystemComponentFromActorInfo();
check(AbilitySystem)
```
전에 if문으로 내부에 변수 선언하는 것은 별로라하셨으니 이 방식은 어떤가요? 이거랑 동일한 기능을 하니 이 방식도 괜찮아보이는데

Reply:
수정했습니다